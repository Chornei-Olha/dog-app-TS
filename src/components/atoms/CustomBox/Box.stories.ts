import type { Meta, StoryObj } from '@storybook/react';

import CustomBox from './Box'