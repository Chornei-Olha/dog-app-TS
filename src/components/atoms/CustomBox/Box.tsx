import { Box, BoxProps } from '@mui/material';
import { styled } from '@mui/material/styles';
import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';

const theme = createTheme();

interface CustomBoxProps extends BoxProps {
  borderRadius?: number;
  height?: number;
  isHovered: boolean;
}

const CustomBox = styled(Box)<CustomBoxProps>(
  ({ theme, borderRadius, height, isHovered }) => ({
    borderRadius: borderRadius || 20,
    height: height || 50,
    boxShadow: isHovered
      ? `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
      : 'none',
    transition: 'box-shadow 0.3s ease-in-out',
    '&:hover': {
      boxShadow: isHovered ? `6px 6px 8px 0px #921FED` : 'none'
    },
    '&:active': {
      boxShadow: isHovered
        ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
        : 'none'
    }
  })
);

export default CustomBox;

