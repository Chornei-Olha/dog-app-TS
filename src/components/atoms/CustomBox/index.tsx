import { Box, BoxProps, alpha } from '@mui/material';
import { styled } from '@mui/material/styles';

interface BorderedBoxProps extends BoxProps {
  borderRadius?: number;
  height?: number;
  showHovered: boolean;
}

const BorderedBox = styled(Box)<BorderedBoxProps>(
  ({ theme, borderRadius, height, showHovered }) => ({
    borderRadius: borderRadius || 20,
    height: height || 50,
    boxShadow: `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`,
    transition: 'box-shadow 0.3s ease-in-out',
    '&:hover': {
      boxShadow: showHovered
        ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
        : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    },
    '&:active': {
      boxShadow: showHovered
        ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
        : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    }
  })
);

export default BorderedBox;
