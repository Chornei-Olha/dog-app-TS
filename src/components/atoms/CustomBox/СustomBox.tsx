import { Box, BoxProps } from '@mui/material';
import { styled } from '@mui/material/styles';
import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';

const theme = createTheme();

interface CustomBoxProps extends BoxProps {
  borderRadius?: number;
  height?: number;
  showHovered: boolean;
}

const CustomBox = styled(Box)<CustomBoxProps>(
  ({ theme, borderRadius, height, showHovered }) => ({
    borderRadius: borderRadius || 20,
    height: height || 50,
    boxShadow: `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`,
    transition: 'box-shadow 0.3s ease-in-out',
    '&:hover': {
      boxShadow: showHovered
        ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
        : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    },
    '&:active': {
      boxShadow: showHovered
        ? `6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
        : `6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    },
    'data-testid': 'custom-box'
  })
);

export default CustomBox;
