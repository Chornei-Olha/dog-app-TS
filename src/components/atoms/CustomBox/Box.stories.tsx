// import type { Meta, StoryObj } from '@storybook/react';

// import CustomBox from './Box'

// const meta = {
//     title: 'CustomBox',
//     parameters: {
//         layout: 'centered'
//       }
// } satisfies Meta<typeof CustomBox>;

// export default meta;

// type Story = StoryObj<typeof meta>;

import React from 'react';
import CustomBox from './Box';

export const withBoxHoverEffect = (Story: React.ComponentType) => {
  return (
    <div
      onMouseEnter={() => <Story isHovered={true} />}
      onMouseLeave={() => <Story isHovered={false} />}
    >
      <Story />
    </div>
  );
};

export default {
  title: 'CustomBox',
  component: CustomBox,
  decorators: [withBoxHoverEffect],
};

export const Default = () => <CustomBox isHovered={false}>Обычный Box</CustomBox>;

// export const Hovered = () => <CustomBox isHovered={true}>Box с hover-эффектом</CustomBox>;



  
  


