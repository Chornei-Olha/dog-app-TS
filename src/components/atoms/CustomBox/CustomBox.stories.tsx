import React from 'react';
import { Meta, Story } from '@storybook/react';
import CustomBox from './Ð¡ustomBox';

import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';

const theme = createTheme();

const withHoverEffect = Story => {
  return (
    <div
      onMouseEnter={() => Story({ showHovered: true })}
      onMouseLeave={() => Story({ showHovered: false })}
    >
      <Story showHovered={false} />
    </div>
  );
};

export default {
  title: 'CustomBox',
  component: CustomBox,
  decorators: [withHoverEffect]
};

export const HoverWithBlackShadow = () => (
  <CustomBox
    showHovered={false}
    boxShadow={alpha(theme.palette.common.white, 1)}
  >
    Box with BLACK shadow on hover
  </CustomBox>
);

export const HoverWithMainShadow = () => (
  <CustomBox
    showHovered={true}
    boxShadow={alpha(theme.palette.primary.main, 1)}
  >
    Box with MAIN shadow on hover
  </CustomBox>
);
