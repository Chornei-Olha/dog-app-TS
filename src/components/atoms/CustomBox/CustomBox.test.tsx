import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/';
import CustomBox from './Ð¡ustomBox';
import { alpha } from '@mui/material';
import { createTheme } from '@mui/material/styles';

const theme = createTheme();

describe('CustomBox', () => {
  it('should render without errors', () => {
    render(<CustomBox data-testid="custom-box" showHovered={false} />);
    expect(screen.getByTestId('custom-box')).toBeInTheDocument();
    expect('custom-box').toMatchSnapshot();
  });

  it('the default state should be a black shadow', () => {
    render(<CustomBox data-testid="custom-box" showHovered={false} />);
    const customBox = screen.getByTestId('custom-box');
    expect(customBox).toHaveStyle(
      `box-shadow: 6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    );
    expect('custom-box').toMatchSnapshot();
  });

  it('should have a black shadow when transmitting this props', () => {
    render(<CustomBox data-testid="custom-box" showHovered={false} />);
    const customBox = screen.getByTestId('custom-box');
    userEvent.hover(customBox);
    expect(customBox).toHaveStyle(
      `box-shadow: 6px 6px 0px 0px ${alpha(theme.palette.common.black, 1)}`
    );
    expect('custom-box').toMatchSnapshot();
  });

  it('should have a main shadow when transmitting this props', () => {
    render(<CustomBox data-testid="custom-box" showHovered={true} />);
    const customBox = screen.getByTestId('custom-box');
    userEvent.hover(customBox);
    expect(customBox).toHaveStyle(
      `box-shadow: 6px 6px 8px 0px ${alpha(theme.palette.primary.main, 1)}`
    );
    expect('custom-box').toMatchSnapshot();
  });
});
