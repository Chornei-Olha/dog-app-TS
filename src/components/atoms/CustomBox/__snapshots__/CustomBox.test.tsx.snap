// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBox should have a black shadow when transmitting this props 1`] = `"custom-box"`;

exports[`CustomBox should have a main shadow when transmitting this props 1`] = `"custom-box"`;

exports[`CustomBox should render without errors 1`] = `"custom-box"`;

exports[`CustomBox the default state should be a black shadow 1`] = `"custom-box"`;
